src/backend/distributed/executor/local_executor.c

```diff
@@ -235,6 +237,146 @@ ExtractParametersForLocalExecution(ParamListInfo paramListInfo, Oid **parameterT
 }
 
 
+/*
+ * ExecuteLocalUtilityTaskList executes a list of tasks locally. This function
+ * also logs local execution notice for each task and sets
+ * TransactionAccessedLocalPlacement to true for next set of possible queries
+ * & commands within the current transaction block. See the comment in function.
+ */
+void
+ExecuteLocalUtilityTaskList(List *localTaskList)
+{
+	Task *localTask = NULL;
+
+	foreach_ptr(localTask, localTaskList)
+	{
+		const char *localTaskQueryCommand = TaskQueryString(localTask);
+
+		if (localTask->anchorShardId != INVALID_SHARD_ID)
+		{
+			/*
+			 * If anchor shard id is valid, then we really access a local
+			 * placement. We should register it to force the local execution
+			 * of the following commands withing the current transaction.
+			 * It can be a coordinated transaction or a transaction that is
+			 * initiated explicitly.
+			 */
+			TransactionAccessedLocalPlacement = true;
+		}
+
+		LogLocalCommand(localTask);
+
+		LocallyExecuteUtilityTask(localTaskQueryCommand);
+	}
+}
+
+
+/*
+ * LocallyExecuteUtilityTask executes the given local task query in the current
+ * session. Given query would mostly be a utility command as the name of the
+ * function implies. However, some utility commands can trigger udf calls to
+ * finish their execution. See the comment in function.
+ */
+static void
+LocallyExecuteUtilityTask(const char *localTaskQueryCommand)
+{
+	RawStmt *localTaskRawStmt = (RawStmt *) ParseTreeRawStmt(localTaskQueryCommand);
+
+	Node *localTaskRawParseTree = localTaskRawStmt->stmt;
+
+	/*
+	 * Actually, the query passed to this function would mostly be a
+	 * utility command to be executed locally. However, some utility
+	 * commands do trigger udf calls (e.g worker_apply_shard_ddl_command)
+	 * to execute commands in a generic way. But as we support local
+	 * execution of utility commands, we should also process those udf
+	 * calls locally as well. In that case, we simply execute the query
+	 * implying the udf call in below conditional block.
+	 */
+	if (IsA(localTaskRawParseTree, SelectStmt) && SelectStmtHasOnlyUdfTargets(
+			(SelectStmt *) localTaskRawParseTree))
```

Comment:
`SelectStmtHasOnlyUdfTargets` seems somewhat arbitrary. e.g. It supports `SELECT a(1), b(2) FROM table`. Why have the check at all?

Reply:
just to be sure that if SelectStmt does only include UDF calls in it

Reply:
otherwise, we do not expect the other kind of SELECT queries to fall here

Reply:
That seems a bit arbitrary. If there is ever a day when someone passes in a SELECT without a UDF in the target list, we'll pass it to CitusProcessUtility, which likely crashes. Maybe we can omit the check to reduce the amount of code and make it more robust?

Reply:
done